### PR TITLE
wine-wayland: 7.0-rc2 -> 7.20

### PR DESCRIPTION
**Pull request by rhendric**
###### Description of changes

https://www.winehq.org/announce/7.20

See also #194215.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by henkery — APPROVED** [review]

File: pkgs/applications/emulators/wine/base.nix [hunk 1]
```diff
@@ -107,7 +107,7 @@ stdenv.mkDerivation ((lib.optionalAttrs (buildScript != null) {
      libX11 libXi libXcursor libXrandr libXrender libXxf86vm libXcomposite libXext
   ])
   ++ lib.optionals waylandSupport (with pkgs; [
-     wayland libxkbcommon wayland-protocols wayland.dev libxkbcommon.dev
+     wayland libxkbcommon wayland-protocols wayland.dev libxkbcommon.dev mesa.dev
```
Comment (Atemu) [inline pkgs/applications/emulators/wine/base.nix:110]: Why is this required?

Comment (rhendric) [reply pkgs/applications/emulators/wine/base.nix:110]: [This header](https://gitlab.collabora.com/alf/wine/-/blob/wayland/dlls/winewayland.drv/waylanddrv.h#L28) includes `<gbm.h>`; I couldn't find a libgbm in nixpkgs, and mesa is the only place in my Nix store that I found a `gbm.h` file. :shrug: Is there a better dependency for me to take?

Comment (Atemu) [reply pkgs/applications/emulators/wine/base.nix:110]: No that's alright then.

Comment (SuperSamus) [reply pkgs/applications/emulators/wine/base.nix:110]: Maybe you should put a comment on it.
Also, a bit OT, searching though nixpkgs there are other packages that use `mesa` just for `gbm` (I found at least 15 of them thanks to comments), however they simply use `mesa` in `buildInputs` instead of `mesa.dev`. Should that be changed?

Comment (Atemu) [reply pkgs/applications/emulators/wine/base.nix]: BuildInputs is magic and automatically chooses the dev input I believe.

We should definitely try and decouple gbm.h from mesa to reduce rebuilds when we try to update the driver.

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 199295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wine-wayland</li>
  </ul>
</details>

Was able to run winecfg.

How do I test that it's actually wayland?

**Comment by rhendric** [comment]
> How do I test that it's actually wayland?

`DISPLAY=:666 winecfg`?

**Review by Atemu — APPROVED** [review]

**Comment by SuperSamus** [comment]
Could #194215 also be merged?
